### DISCUSSION TITLE
Get json_pointer from json object?

### DISCUSSION THREAD

**Discussion by clarkholdham** [nlohmann/json#3508] [Q&A]
Is there anyway to get a json_pointer from a json object? Basically I would like to store json_pointers string representation to access the same nested object later. I know that ptr has the concept of it's parent so I was hoping the pointer could be reconstructed using the parents.

```
nlohmann::json j = R"###({
    "parent": {
        "child": {
            "data": 1
        }
    }
})###"_json;

auto child = j.at("parent").at("child");
// auto child = j.at("/parent/child");

// this is the function I don't think exists
nlohmann::json::json_pointer ptr = child.get_json_pointer();

// now access again using this ptr
nlohmann::json child_ = j.at(ptr.to_string());

if(child == child_) {
    std::cout << "Same same";
}
```

**Comment by nlohmann** [comment]
The problem is that `child` does not know anything about its parent. We fixed this by adding a pointer for the extended diagnostic messages (https://json.nlohmann.me/home/exceptions/#extended-diagnostic-messages). It *could* be possible to create such a function as you describe it. The problem is that it would only be available in case `JSON_DIAGNOSTICS` is defined (see https://json.nlohmann.me/api/macros/json_diagnostics/).

**Reply by clarkholdham** [reply]
We happen to have `JSON_DIAGNOSTICS` enabled right now. Is the parent information internal or could I make a function using the current API?

**Reply by nlohmann** [reply]
Right now it's internal as it is an implementation detail for the extended diagnostics.

**Reply by clarkholdham** [reply]
Thanks for the help.

**Comment by scherer-michael** [comment]
Hi !
I have currently the same need...
The subject is to create a json document referencing objects, added in another json file (instead of having data duplication in both files).
I need to ask : do you think there is any way to retrieve the json_pointer of a json object, with the current version of the library (v3.11.2) ?
Thank in advance for any help or advice.

**Reply by scherer-michael** [reply]
I tried a hack-isch way like the following, but obviously, the returned path is not really helpful... Secondly, for json with a huge amount of objects, this leads to a huge amount of processing.
```cpp
int main() {
    nlohmann::json j = R"###({
        "list": [
            {
                "data": 1
            },
            {
                "data": 1
            }
        ]
    })###"_json;

    auto before = j;

    nlohmann::json chunck = R"###(
            {
                "data": 1
            })###"_json;

    j.at("list").push_back(chunck);

    auto diff = nlohmann::json::diff(before, j);

    fmt::print("{}", diff.dump(4));
}
// Output
//[
//    {
//        "op": "add",
//        "path": "/list/-",
//        "value": {
//            "data": 1
//        }
//    }
//]
```
